modules/fedimint-ln-client/src/lib.rs

```diff
@@ -171,24 +213,31 @@ impl LightningClientExt for Client {
         Ok(instance.api.fetch_gateways().await?)
     }
 
-    async fn pay_bolt11_invoice(
-        &self,
-        invoice: Invoice,
-    ) -> anyhow::Result<(OperationId, ContractId)> {
+    async fn pay_bolt11_invoice(&self, invoice: Invoice) -> anyhow::Result<(PayType, ContractId)> {
         let (lightning, instance) = self.get_first_module::<LightningClientModule>(&KIND);
-        let operation_id = OperationId(invoice.payment_hash().into_inner());
-        let active_gateway = self.select_active_gateway().await?;
+        let payment_hash = invoice.payment_hash();
+        let operation_id = OperationId(payment_hash.into_inner());
 
-        let (output, contract_id) = lightning
-            .create_outgoing_output(
-                operation_id,
-                instance.api,
-                invoice.clone(),
-                active_gateway,
-                self.get_config().await.federation_id,
-                rand::rngs::OsRng,
-            )
-            .await?;
+        let gateways = self.fetch_registered_gateways().await?;
+        let (pay_type, output, contract_id) = if check_internal_payment(&invoice, gateways).await {
```

Comment:
nit: I prefer `is_internal_payment` so its obvious what the function is returning

Reply:
makes sense. Updated